src/domain/user-group/user-group.service.ts

```diff
@@ -73,10 +74,71 @@ export class UserGroupService {
         LogContext.COMMUNITY
       );
 
+    // Cannot remove restricted groups
+    if (await this.isRestricted(group))
+      throw new ForbiddenException(
```

Comment:
Forbidden = Unauthorized. This is unrelated to auth issues. Use ValidationException instead.